packages/cosmic-swingset/src/launch-chain.js

```diff
@@ -724,33 +731,16 @@ export async function launch({
 
   /**
    * @template T
-   * @param {string} type
+   * @param {string} label
    * @param {() => Promise<T>} fn
+   * @param {() => void} onSettled
    */
-  async function processAction(type, fn) {
-    const start = Date.now();
-    const finish = res => {
-      // blockManagerConsole.error(
-      //   'Action',
-      //   action.type,
-      //   action.blockHeight,
-      //   'is done!',
-      // );
-      runTime += Date.now() - start;
-      return res;
-    };
-
+  function withErrorLogging(label, fn, onSettled) {
     const p = fn();
-    // Just attach some callbacks, but don't use the resulting neutered result
-    // promise.
-    void E.when(p, finish, e => {
-      // None of these must fail, and if they do, log them verbosely before
-      // returning to the chain.
-      blockManagerConsole.error(type, 'error:', e);
-      finish();
+    void E.when(p, onSettled, err => {
```

Comment:
I'm surprised that the previous `finish` took care of returning `res` just for getting `void`ed here. We probably should review the whole composition of this thing.